### DISCUSSION TITLE
does not have an associated "Action" error

### DISCUSSION THREAD

**Discussion by Ali-toghraee** [idaholab/moose#16707] [Q&A Modules: Phase field] [answered]
Dear MOOSE users,

I have already  had an input file to model oxidation which incorporates AC+CH+Tensor Mech. 
Now, I want to modify it to have a bi-crystal model. To add the the Grain Growth equation   only on my metal phase (grains)
I tried to add "ACGrGrPoly" and "ACInterface" kernels only to my metal phase (grains) eta_Fe and eta_Fe2. However, when I run the input I receive an error message of  _"section 'ACBulk00' does not have an associated "Action"._ 

Can anybody help on this one?

I have attached the input file.

[4eta_bi_grgr_forgithub.txt](https://github.com/idaholab/moose/files/5818667/4eta_bi_grgr_forgithub.txt)

**Comment by GiudGiud** [comment, marked as answer]
Hello

From your input file it looks like the kernel blocks you added (ACGrGrPoly ACinterface) are after the [] closing the Kernels. So MOOSE is searching for an action, rather than a kernel. They need to be moved up inside the
[Kernels]
...
[] 
structure. 

Best,
Guillaume

**Reply by Ali-toghraee** [reply]
Oh, Thanks for catching that Guillaume!
